api/controllers/game/stalemate.js

```diff
@@ -25,30 +24,36 @@ module.exports = async function (req, res) {
     };
 
     gameUpdates.lastEvent = { change: 'requestStalemate', requestedByPNum: pNum };
-
     // End in stalemate if both players requested stalemate this turn
     if (playerStalemateVal === opponentStalemateVal && opponentStalemateVal === game.turn) {
-      victory.gameOver = true;
       gameUpdates = {
         ...gameUpdates,
         p0: game.players[0].id,
         p1: game.players[1].id,
         status: gameService.GameStatus.FINISHED,
         winner: null,
       };
-      updatePromises.push(gameService.clearGame({ userId }));
-    }
-
-    updatePromises.push(Game.updateOne({ id: gameId }).set(gameUpdates));
-    await Promise.all(updatePromises);
-
-    if (victory.gameOver && gameUpdates.status === gameService.GameStatus.FINISHED) {
+      
+      await Game.updateOne({ id: gameId }).set(gameUpdates);
+      game = await gameService.populateGame({ gameId }); 
+      
+      victory.gameOver = true;
       victory.currentMatch = await sails.helpers.addGameToMatch(game);
+      
+      await Game.updateOne({ id: gameId }).set({
+        lastEvent: {
+          change: 'requestStalemate',
+          game: game,
+          victory
+        }
+      });      
+      await gameService.clearGame({ userId });
+    } else {
+      await Game.updateOne({ id: gameId }).set(gameUpdates);
     }
-
     Game.publish([game.id], {
       change: 'requestStalemate',
-      game,
+      game: game,
```

Comment:
This is the equivalent shorthand for the previously existing line
```suggestion
      game,
```